### DISCUSSION TITLE
Applying SingleR to GEX+ADT data from CITE-seq

### DISCUSSION THREAD

**Discussion by PeteHaitch** [LTLA/SingleR#242] [General]
I've been playing around with ways to use **SingleR** on CITE-seq datasets.

I've started with the 'Azimuth Human - PBMC' dataset (https://azimuth.hubmapconsortium.org/references/#Human%20-%20PBMC).
As expected, there's a lot of drudgery and rabbit holes to go down just in trying to get the data into a usable form.
Then it's how best to combine the GEX and ADT data and parameter choices for the purpose of training (i.e. with `trainSingleR()`).

This is work in progress and I can post back with code and results, but I'm wondering if others have used **SingleR** to train a model on similar reference datasets or otherwise thought about it?

The long term idea would be to make some well-annotated CITE-seq datasets (perhaps those from https://azimuth.hubmapconsortium.org/references/) available through **celldex** and for them to be as easy to use with `SingleR()` as the current reference datasets.

**Comment by LTLA** [comment]
> This is work in progress and I can post back with code and results, but I'm wondering if others have used **SingleR** to train a model on similar reference datasets or otherwise thought about it?

I have thought about it briefly. I suppose there's no reason that it shouldn't work; IIRC the only issue was that we couldn't be guaranteed the same set of protein features, but if people have finally agreed on a panel, then it might work pretty well.

Probably wouldn't even need all of **SingleR**'s fancy stuff, ADT signal seems so black and white whenever I look at it.

> The long term idea would be to make some well-annotated CITE-seq datasets (perhaps those from https://azimuth.hubmapconsortium.org/references/) available through celldex)

This would be a good idea. I've been wanting to get more references in **celldex** for a while, but it's hard for others to contribute. Maybe life will be easier once we have an **ArtifactDB** rolled out.

In any case, it seems that there would be two objects for each single-cell reference; the full reference, for posterity's sake, and some aggregated reference generated with `aggregateReference()` for quick-and-dirty usage.
